src/control/InstantaneousTrajectoryExecutor.cpp

```diff
@@ -40,9 +44,12 @@ void InstantaneousTrajectoryExecutor::validate(
     throw std::invalid_argument(
         "Trajectory is not in a MetaSkeletonStateSpace.");
 
-  // TODO: Delete this line once the skeleton is locked by isCompatible
-  std::lock_guard<std::mutex> lock(mSkeleton->getMutex());
-  space->checkIfContained(mSkeleton.get());
+  // Check that traj space is compatible with metaskeleton
+  if (!space->isCompatible(mMetaSkeleton.get()))
+  {
+    throw std::invalid_argument(
+        "Trajectory StateSpace incompatible with MetaSkeleton");
```

Comment:
@egordon FYI this doesn't work on gen2 real. Specifically, with `ada_feeding` on `emprise/gen3-integration`, `libada` on `egordon/executor_hotfix`, and `aikido` on `egordon/executor`, this exception gets called the first time the arm tries to execute (in moveAbovePlate).